resource/kind.go

```diff
@@ -0,0 +1,319 @@
+package resource
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	_ Kind    = &UntypedKind{}
+	_ Object2 = &UntypedObject{}
+)
+
+// Object2 implements kubernetes' runtime.Object and meta/v1.Object, as well as some additional methods useful for the app-sdk
+type Object2 interface {
+	runtime.Object
+	metav1.Object
+	GetSpec() any
+	GetSubresources() map[string]any
+	GetStaticMetadata() StaticMetadata
+	SetStaticMetadata(metadata StaticMetadata)
+}
+
+// MultiVersionKind is a collection of Kinds which have the same Group and Kind, but different versions.
+// This may allow us to include conversion as a component of the interface.
+// TODO: not sure if necessary
+type MultiVersionKind interface {
+	Group() string
+	Kind() string
+	Versions() []Kind
+}
+
+// Group is a collection of Kinds which share a Group and Version, allowing them to be put under the same
+// routing in an API server.
+// TODO: not sure if necessary
+type Group interface {
+	Group() string
+	Version() string
+	Kinds() []Kind
```

Comment:
Hm, shouldn't this be `[]MultiVersionKind`? Or maybe even `map[string]MultiVersionKind`?